Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -4752,78 +4752,80 @@ void CPhysicalGeometry::SetPoint_Connectivity() {
 
 void CPhysicalGeometry::SetRCM_Ordering(CConfig *config) {
 
-  queue<unsigned long> Queue;
-  vector<char> inQueue(nPoint, false);
+  /*--- The result is the RCM ordering, during the process it is also used as
+   * the queue of new points considered by the algorithm. This is possible
+   * because points move from the front of the queue to the back of the result,
+   * which is equivalent to incrementing an integer marking the end of the
+   * result and the start of the queue. ---*/
+  vector<char> InQueue(nPoint, false);
   vector<unsigned long> AuxQueue, Result;
   Result.reserve(nPoint);
+  unsigned long QueueStart = 0;
 
-  /*--- Select the node with the lowest degree in the grid. ---*/
-
-  unsigned long AddPoint = 0;
-  auto MinDegree = nodes->GetnPoint(AddPoint);
-  for (auto iPoint = 1ul; iPoint < nPointDomain; iPoint++) {
-    auto Degree = nodes->GetnPoint(iPoint);
-    if (Degree < MinDegree) { MinDegree = Degree; AddPoint = iPoint; }
+  /*--- Exclude halo nodes from the ordering process. ---*/
+  for (auto iPoint = nPointDomain; iPoint < nPoint; iPoint++) {
+    InQueue[iPoint] = true;
   }
 
-  /*--- Add the node in the first free position. ---*/
+  /*--- Repeat as many times as necessary to handle disconnected graphs. ---*/
+  while (Result.size() < nPointDomain) {
```

Comment:
Reason for the residual changes.

Reply:
-> (from dev meeting) Halo points should be at the end of the point list as a convention.
The disconnected domain parts were the change (as they happen in the CHT pin case where disconnected pin-pieces are treated as 1 zone)